src/devtools/client/webconsole/utils/autocomplete.ts

```diff
@@ -0,0 +1,249 @@
+  properties: ObjectPreviewProperty[];
+  prototypeId: any;
+  prototypeValue: ValueFront;
+};
+type ObjectFront = {
+  className: string;
+  objectId: string;
+  preview?: ObjectPreview;
+};
+type ValueFront = {
+  _pause: any;
+  _hasPrimitive: false;
+  _primitive?: boolean;
+  _isBigInt: boolean;
+  _isSymbol: boolean;
+  _object: ObjectFront;
+  _uninitialized: boolean;
+  _unavailable: boolean;
+  _elements: null;
+  _isMapEntry: null;
+};
+type Binding = {
+  name: string;
+  value: ValueFront;
+};
+type Scope = {
+  bindings: Binding[];
+  actor: string;
+  parent: Scope;
+  functionName?: string;
+  object?: ValueFront;
+  scopeKind: string;
+  type: "block" | string;
+};
+type FrameScope = {
+  scope: Scope;
+  pending: boolean;
+  originalScopesUnavailable: boolean;
+};
+
+function getMatchString(str: string) {
+  return str.toLowerCase().replace(/['"`]+/g, "");
+}
+function fuzzyFilter(candidates: string[], query: string): string[] {
+  if (getMatchString(query) === "") {
+    return candidates;
+  }
+  return filter(candidates, query);
+}
+
+function isIdentifier(expression: string) {
+  try {
+    const parsed = babelParser.parseExpression(expression);
+    const isIdentifier = t.isIdentifier(parsed);
+
+    return isIdentifier;
+  } catch (e) {
+    return false;
+  }
+}
+function isMemberExpression(expression: string) {
+  try {
+    const parsed = babelParser.parseExpression(expression);
+    const rv = t.isMemberExpression(parsed);
+
+    return rv;
+  } catch (e) {
+    return false;
+  }
+}
+
+function isBracketNotation(input: string) {
+  const expression = input.slice(0, input.lastIndexOf("[") + 1);
+  return isMemberExpression(expression + `"a"]`);
+}
+function isDotNotation(input: string) {
+  return isMemberExpression(input + `a`);
+}
+function isAccessingObjProperty(input: string) {
+  return isBracketNotation(input) || isDotNotation(input);
+}
+
+function getPropertiesForObject(object: ObjectFront): string[] {
+  const preview = object.preview;
+  const properties = [];
+
+  // The object preview may be undefined until the user expands the object
+  // in the scopes panel. This applies primarily to prototype objects.
+  if (preview) {
+    const objectProperties = preview.properties.map(b => b.name);
+    properties.unshift(...objectProperties);
+  } else {
+    if (["Object", "Array"].includes(object.className)) {
+      const prototypeProperties = Object.getOwnPropertyNames(Object.prototype);
+      properties.unshift(...prototypeProperties);
+    }
+
+    if (object.className === "Array") {
+      const prototypeProperties = Object.getOwnPropertyNames(Array.prototype);
+      properties.unshift(...prototypeProperties);
+    }
+  }
+
+  const prototype = preview?.prototypeValue;
+
+  // Recursively gather the properties through the prototype chain.
+  if (prototype?._object) {
+    const prototypeProperties = getPropertiesForObject(prototype._object);
+    properties.unshift(...prototypeProperties);
```

Comment:
Oooooooooooo this is fun. OK, if what I said above gets way harder then maybe don't worry about it 😄 .

But it seems to me like 
```
return [...getLocalProperties(object), ...getLocalProperties(object.preview?.prototypeValue?._object)]```

might work?